public/app/features/dashboard-scene/serialization/transformSaveModelSchemaV2ToScene.ts

```diff
@@ -249,12 +255,90 @@ function buildGridItem(gridItem: GridLayoutItemSpec, panel: PanelKind, yOverride
   });
 }
 
-function createSceneGridLayoutForItems(dashboard: DashboardV2Spec): SceneGridItemLike[] {
-  const gridElements = dashboard.layout.spec.items;
+function createLayoutManager(dashboard: DashboardV2Spec): DashboardLayoutManager {
+  if (dashboard.layout.kind === 'GridLayout') {
+    return new DefaultGridLayoutManager({
+      grid: new SceneGridLayout({
+        isLazy: !(dashboard.preload || contextSrv.user.authenticatedBy === 'render'),
+        children: createSceneGridLayoutForItems(dashboard.layout, dashboard.elements),
+        $behaviors: [trackIfEmpty],
+      }),
+    });
+  } else if (dashboard.layout.kind === 'RowsLayout') {
+    return new RowsLayoutManager({
+      rows: dashboard.layout.spec.rows.map((row) => {
+        let layout: DashboardLayoutManager | undefined = undefined;
+
+        if (row.spec.layout.kind === 'RowGridLayout') {
+          layout = new DefaultGridLayoutManager({
+            grid: new SceneGridLayout({
+              children: createSceneGridLayoutForItems(row.spec.layout, dashboard.elements),
+            }),
+          });
+        }
+
+        if (row.spec.layout.kind === 'ResponsiveGridLayout') {
+          layout = new ResponsiveGridLayoutManager({
+            layout: new SceneCSSGridLayout({
+              templateColumns: row.spec.layout.spec.col,
+              autoRows: row.spec.layout.spec.row,
+              children: createResponsiveGridItems(row.spec.layout.spec.items, dashboard.elements),
+            }),
+          });
+        }
+
+        if (!layout) {
+          throw new Error(`Unsupported layout kind: ${row.spec.layout.kind} in row`);
+        }
+        return new RowItem({
+          title: row.spec.title,
+          isCollapsed: row.spec.collapsed,
+          layout: layout,
+        });
+      }),
+    });
+  } else if (dashboard.layout.kind === 'ResponsiveGridLayout') {
+    return new ResponsiveGridLayoutManager({
+      layout: new SceneCSSGridLayout({
+        templateColumns: dashboard.layout.spec.col,
+        autoRows: dashboard.layout.spec.row,
+        children: createResponsiveGridItems(dashboard.layout.spec.items, dashboard.elements),
+      }),
+    });
+  }
+
+  // @ts-ignore - this complains because we should never reach this point. If the model does not match the schema we will though.
+  throw new Error(`Unsupported layout type: ${dashboard.layout.kind}`);
+}
+
+function createResponsiveGridItems(
+  items: ResponsiveGridLayoutItemKind[],
+  elements: Record<string, Element>
+): ResponsiveGridItem[] {
+  return items.map((item) => {
+    const panel = elements[item.spec.element.name];
+    if (!panel) {
+      throw new Error(`Panel with uid ${item.spec.element.name} not found in the dashboard elements`);
+    }
+    if (panel.kind !== 'Panel') {
+      throw new Error(`Unsupported element kind: ${panel.kind}`);
+    }
+    return new ResponsiveGridItem({
+      key: `grid-item-${panel.spec.id}`,
```

Comment:
Nit: We have a helper method for creating those keys: `getGridItemKeyForPanelId`